script.ebooks/plugin.py

```diff
@@ -1,568 +1,600 @@
-        cover = args.get('cover', None)
-
-        if (cover is not None) and (len(cover) > 0):
-            cover = cover[0]
-        else:
-            cover = None
-
-        if (filename is not None) and (len(filename) > 0):
-            menuNav = MenuNavigator(base_url, addon_handle)
-            menuNav.listChapters(filename[0], cover)
-            del menuNav
-
-    elif mode[0] == 'readChapter':
-        log("EBooksPlugin: Mode is READ CHAPTER")
-
-        # Get the actual chapter that was navigated to
-        filename = args.get('filename', None)
-        link = args.get('link', None)
-        title = args.get('title', None)
-        isFirstChapterVal = args.get('firstChapter', None)
-        isLastChapterVal = args.get('lastChapter', None)
-
-        if (title is not None) and (len(title) > 0):
-            title = title[0]
-        else:
-            title = ""
-
-        isFirstChapter = False
-        if (isFirstChapterVal is not None) and (len(isFirstChapterVal) > 0):
-            if isFirstChapterVal[0] == 'true':
-                isFirstChapter = True
-        isLastChapter = False
-        if (isLastChapterVal is not None) and (len(isLastChapterVal) > 0):
-            if isLastChapterVal[0] == 'true':
-                isLastChapter = True
-
-        if (filename is not None) and (len(filename) > 0) and (link is not None) and (len(link) > 0):
-            menuNav = MenuNavigator(base_url, addon_handle)
-            menuNav.readChapter(filename[0], title, link[0], isFirstChapter, isLastChapter)
-            del menuNav
-
-    elif mode[0] == 'markReadStatus':
-        log("EBooksPlugin: Mode is MARK READ STATUS")
-
-        filename = args.get('filename', None)
-        link = args.get('link', None)
-        readStatus = args.get('read', None)
-
-        if (link is not None) and (len(link) > 0):
-            link = link[0]
-        else:
-            link = ""
-
-        markRead = False
-        if (readStatus is not None) and (len(readStatus) > 0):
-            if readStatus[0] == 'true':
-                markRead = True
-
-        if (filename is not None) and (len(filename) > 0):
-            menuNav = MenuNavigator(base_url, addon_handle)
-            menuNav.markReadStatus(filename[0], link, markRead)
-            del menuNav
-
-    elif mode[0] == 'opds':
-        log("EBooksPlugin: Mode is OPDS")
-
-        menuNav = MenuNavigator(base_url, addon_handle)
-        menuNav.opdsRootMenu()
-        del menuNav
-
-    elif mode[0] == 'opds2':
-        log("EBooksPlugin: Mode is OPDS 2")
-
-        href = args.get('href', None)
-
-        if (href is not None) and (len(href) > 0):
-            menuNav = MenuNavigator(base_url, addon_handle)
-            menuNav.opdsBookListing(href[0])
-            del menuNav
+# -*- coding: utf-8 -*-
+import sys
+import os
+import urllib
+import urlparse
+import xbmc
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import xbmcvfs
+
+__addon__ = xbmcaddon.Addon(id='script.ebooks')

```

Comment:
We are trying not use the label format _\_name_\_ it is used for Python internals and is confusing.